src/App.php

```diff
@@ -158,36 +147,58 @@ public function dispatch(): ResponseInterface
                 $controllerName = $routeInfo[1];
                 if (isset($routeInfo[2])) {
                     foreach ($routeInfo[2] as $name => $value) {
-                        $this->request = $this->request->withAttribute($name, $value);
+                        $request = $request->withAttribute($name, $value);
                     }
                 }
                 break;
         }
         // default HTTP response
         $response = new Response(200);
 
-        $controllerName = is_array($controllerName) ?: [$controllerName];
-        foreach ($controllerName as $controller) {
-            $this->logger->debug(sprintf("Executing %s", $controller));
+        if (!is_array($controllerName)) {
```

Comment:
Super minor note:

`!is_array(null)` === `true`.

It can't happen, I know, but we never know the next implementation of FastRoute.

Reply:
I tested using `$controllerName = null` and it seems to generate an error, no side effects. I think we can keep this as is, at least for now.